### ISSUE TITLE
ASSINATURA DIGITAL

### ISSUE THREAD

**Issue by paulot12** [interlegis/sapl#969]
Será possível ter o campo para a assinatura digital das proposições, pareceres e todos os documentos legislativos, no SAPL 3.1?

**Comment by edwardoliveira** [comment]
**Infelizmente**, não dispomos desta funcionalidade. A implementação de assinatura digital é um projeto de considerável complexidade, pois envolve não somente a parte de codificação, mas também a infraestrutura implantada. Desta forma, não iremos priorizar esta tarefa tão cedo, mas *definitivamente* faz parte de nosso planos futuros. Só não podemos prometer uma data.

Se **você** ou alguém mais quiser propor uma solução dentro do SAPL e submeter como Pull Request teremos enorme prazer em integrá-lo ao sistema o mais rápido possível. Ajudas são sempre bem vindas!

**Comment by DtiLegBG** [comment]
Uma idéia bacana é seguir o que foi feito nesse fork do sapl
http://sapl.camaraassis.sp.gov.br/consultas/materia/materia_mostrar_proc?cod_materia=NzMzNTM=

Para a assinatura de documentos, pelo que parece, seria interessante avaliar como o sei do trf4 utiliza, e também ferramentas como o shodo, que parecem se destacar.

No caso de uso do SEI, o interessante que os documentos não precisam necessariamente ser assinados por um certificado icp, também podem ser assinados por usuário e senha, para usuários do orgão.

Assinatura e envio eletrônico, essa será a nova grande conquista para o sapl

**Comment by DtiLegBG** [comment]
um exemplo de documento assinado eletronicamente, utilizando usuário e senha,  pelo sistema SEI, desenvolvido pelo trf4 e posterior colocado como texto integrar no SAPL.
Desenvolver essa funcionalidade de assinatura eletrônica dentro do SAPL seria fantástico.
http://sapl.cmm.pr.gov.br:8080/sapl/sapl_documentos/materia/27277_texto_integral

**Comment by edwardoliveira** [comment]
Mais uma sugestão de assinatura informada via OSTicket:

"Gostaria de sugerir a implementação de uma assinatura de documentos pelo próprio SAPL, baseado no login do usuário, semelhante ao Sistema Eletrônico de Informações (SEI). Seria interessante que o SAPL permitisse a criação do documento administrativo ou de uma matéria legislativa e a inserção de um selo de assinatura com os dados dos usuários, semelhante ao processo realizado pelo SEI."

**Comment by edwardoliveira** [comment]
@DtiLegBG 
> seria interessante avaliar como o sei do trf4 utiliza

Isso nos interessa muito, mas o código do SEI do TRF4 é fechado. Tem que assinar um documento de confiencialidade, e não pode divulgar e etc. Iriamos oferecê-lo como produto suportado pelo Interlegis, mas a coisa não avançou muito por conta disso. Eu tenho interesse me ver como eles fizeram, mas não temos acesso ao código fonte.

**Comment by DtiLegBG** [comment]
Olha que interessante essa inciativa do TRF2
https://github.com/projeto-siga/siga

Eles tem implementado a questão de assinatura por certificado, por usuário e senha, verificação de autenticidade.

Ainda sobre o Siga-Doc, um exemplo de uso.
http://www.saopaulo.sp.leg.br/blog/gestao-digital-pad-reduz-consumo-de-papeis/

**Comment by edwardoliveira** [comment]
<img width="1027" alt="Screen Shot 2020-05-13 at 13 14 05" src="https://user-images.githubusercontent.com/9326037/81837787-aac64e80-951b-11ea-9ffb-494716eb30a1.png">

Isso aqui vai ser descontinuado pela Oracle, se já não foi. Precisamos de uma solução que **não use** applets Java, de preferência uma solução em Python que possamos integrar.

**Comment by aloysioy** [comment]
> Olha que interessante essa inciativa do TRF2
> https://github.com/projeto-siga/siga
> 
> Eles tem implementado a questão de assinatura por certificado, por usuário e senha, verificação de autenticidade.
> 
> Ainda sobre o Siga-Doc, um exemplo de uso.
> http://www.saopaulo.sp.leg.br/blog/gestao-digital-pad-reduz-consumo-de-papeis/

Pessoal, que tal se utilizarmos o sigadoc via webseevice e ele fazer o processo eletrônico as assinaturas e tudo mais. Vejo que desta forma fica bom. Estou estudando a implantação do SIGADOC com mysql e devemos entrar em produção em abril, e  estou iniciando a parto do legislativo, no meu ver uma integração entre estes 2 sistemas seria fantástico

**Comment by edwardoliveira** [comment]
Sim, estamos abertos a sugestões de integração e uso do SAPL com outros sistemas. Quanto mais colaboração da comunidade tiver, melhor.

**Comment by cmsbc-aloysio** [comment]
Infelizmente meu projeto atual é fazer o Processo Eletrônico com o SIGADOC, mas assim que começar a andar com as próprias pernas o projeto e eu vou retomar o sistema legislativo aqui, com certeza irei analisar com todo o cuidado uma possível implementação de integração, se funcionar qualquer câmara poderá ter o sistema legislativo e o processo eletrônico gratuitamente.
E o siga está no docker também o banco é diferente é o MySQL mas tudo bem até aí.
